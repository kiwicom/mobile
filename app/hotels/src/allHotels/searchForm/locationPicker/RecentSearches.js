// @flow

import * as React from 'react';
import { View, FlatList, Keyboard } from 'react-native';
import { StyleSheet } from '@kiwicom/mobile-shared';
import { Translation } from '@kiwicom/mobile-localization';

import RecentSearchItem from './RecentSearchItem';

const styles = StyleSheet.create({
  container: {
    padding: 10,
  },
});

type Location = {|
  id: string,
  name: string,
|};

type Props = {|
  locations: Location[],
  onCitySelected: (cityId: string, cityName: string) => void,
|};

export default class RecentSearches extends React.Component<Props> {
  keyExtractor = (item: Location) => item.id;

  renderItem = ({ item }: {| item: Location |}) => (
    <RecentSearchItem
      location={item}
      onCitySelected={this.props.onCitySelected}
    />
  );

  render = () => {
    if (this.props.locations.length === 0) {
      return null;
    }

    return (
      <React.Fragment>
        <View style={styles.container}>
          <Translation id="hotels_search.location_picker.recent_search" />
        </View>
        <FlatList
          data={this.props.locations}
          keyExtractor={this.keyExtractor}
          renderItem={this.renderItem}
          onScroll={Keyboard.dismiss}
          keyboardShouldPersistTaps="handled"
        />
      </React.Fragment>
    );
  };
}
