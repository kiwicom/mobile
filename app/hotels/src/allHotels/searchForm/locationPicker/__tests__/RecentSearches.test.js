// @flow

import * as React from 'react';
import { PlaygroundRenderer } from '@kiwicom/react-native-app-playground';

import RecentSearches from '../RecentSearches';

const locations = [
  {
    id: '1',
    name: 'Rome',
  },
  {
    id: '2',
    name: 'Oslo',
  },
];

describe('RecentSearches', () => {
  it('renders', () => {
    PlaygroundRenderer.render(
      <RecentSearches locations={locations} onCitySelected={jest.fn()} />,
    );
  });

  it('renders correctly with no locations', () => {
    PlaygroundRenderer.render(
      <RecentSearches locations={[]} onCitySelected={jest.fn()} />,
    );
  });
});
