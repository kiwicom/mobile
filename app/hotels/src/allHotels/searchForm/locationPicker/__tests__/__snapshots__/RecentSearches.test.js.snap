// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentSearches renders 1`] = `
<React.Fragment>
  <Component
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Translation
      id="hotels_search.location_picker.recent_search"
    />
  </Component>
  <FlatList
    data={
      Array [
        Object {
          "id": "1",
          "name": "Rome",
        },
        Object {
          "id": "2",
          "name": "Oslo",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyboardShouldPersistTaps="handled"
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    onScroll={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</React.Fragment>
`;

exports[`RecentSearches renders correctly with no locations 1`] = `null`;
